src/algebraic_geometry/morphisms/basic.lean

```diff
@@ -12,6 +12,46 @@ import category_theory.morphism_property
 
 We provide the basic framework for talking about properties of morphisms between Schemes.
 
+A `morphism_property Scheme` is a predicate on morphisms between schemes, and an
+`affine_target_morphism_property` is a predicate on morphisms into affine schemes. Given a
+`P : affine_target_morphism_property`, we may construct a `morphism_property` that holds for
+`f : X ⟶ Y` whenever `P` holds for the restriction of `f` on every affine open subset of `Y`.
+
+## Main definitions
+
+- `algebraic_geometry.affine_target_morphism_property.is_local`: We say that `P.is_local` if `P`
+satisfies the assumptions of the affine communication lemma
+(`algebraic_geometry.of_affine_open_cover`). That is,
+1. `P` respects isomorphisms.
+2. If `P` holds for `f : X ⟶ Y`, then `P` holds for `f ∣_ Y.basic_open r` for any
+  global section `r`.
+3. If `P` holds for `f ∣_ Y.basic_open r` for all `r` in a spanning set of the global sections,
+  then `P` holds for `f`.
+
+- `algebraic_geometry.property_is_local_at_target`: We say that `property_is_local_at_target P` for
+`P : morphism_property Scheme` if
+1. `P` respects isomorphisms.
+2. If `P` holds for `f : X ⟶ Y`, then `P` holds for `f ∣_ U` for any `U`.
+3. If `P` holds for `f ∣_ U` for an open cover `U` of `Y`, then `P` holds for `f`.
+
+## Main results
+
+- `algebraic_geometry.affine_target_morphism_property.is_local.affine_open_cover_tfae`:
+  If `P.is_local`, then `target_affine_locally P f` iff there exists an affine cover `{ Uᵢ }` of `Y`
```

Comment:
This seems to be the first place where `target_affine_locally` is mentioned. I think it deserves a separate introduction.